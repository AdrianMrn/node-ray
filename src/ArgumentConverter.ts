import prettyFormat from 'pretty-format';

export class ArgumentConverter
{
    public static convertToPrimitive(arg: any): any
    {
        if (arg === null) {
            return null;
        }

        if (typeof arg === 'string') {
            return arg;
        }

        if (typeof arg === 'number') {
            return arg;
        }

        if (typeof arg === 'boolean') {
            return arg;
        }

        return ArgumentConverter.prettyFormatForHtml(arg);
    }

    public static prettyFormatForHtml(arg: any): string
    {
        const formatted = prettyFormat(arg, { indent: 4 })
            // format whitespace for display in html
            .replace(/ /g, '&nbsp;')
            .replace(/\r\n|\r|\n/g, '<br>')
            // highlight quoted strings
            .replace(
                /("[^"]+")/g,
                '<code style="font-size: 0.8rem!important;" class="bold text-blue-400 p-0">$1</code>'
            )
            // highlight array contents
            .replace(
                /Array(&nbsp;|\s+)+(\[[^\]]+\])/g,
                '<code style="font-size: 0.8rem!important;" class="text-gray-500 p-0">Array$1$2</code>'
            )
            // highlight types like [Function Abc]
            .replace(/^(\[[^\]]+\])$/g, '<code style="font-size: 0.8rem!important;"class="text-gray-500 p-0">$1</code>')
            // highlight object contents
            .replace(/(\{.+\})/g, '<code style="font-size: 0.8rem!important;" class="text-gray-500 ">$1</code>')
            // highlight keywords
            .replace(
                /(Array|Object|Function|Circular|Symbol|WeakMap|Map)/g,
                '<span style="font-size: 0.8rem!important;" class="text-yellow-400">$1</span>'
            )
            // highlight special chars
            .replace(
                /(:&nbsp;|[,[\]{}])/g,
                '<span style="font-size: 0.8rem!important;" class="text-white">$1</span>'
            );

        return `<code style="font-size: 0.8rem!important;">${formatted}</code>`;
    }
}
