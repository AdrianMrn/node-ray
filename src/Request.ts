import { Payload } from '@/Payloads/Payload';

export class Request {
    protected uuid: string;
    protected payloads: Payload[];
    protected meta: any[];

    public constructor(uuid: string, payloads: Payload[], meta: any[] = []) {
        this.uuid = uuid;
        this.payloads = payloads;
        this.meta = meta;
    }

    public toArray(): Record<string, unknown> {
        return {
            uuid: this.uuid,
            payloads: this.payloads.map(payload => payload.toArray()),
            meta: this.meta,
        };
    }

    public toJson(): string {
        return JSON.stringify(this.toArray());
    }
}
