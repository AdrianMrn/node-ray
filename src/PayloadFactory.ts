/* eslint-disable @typescript-eslint/ban-types */

import { ArgumentConverter } from './ArgumentConverter';
import { BoolPayload } from './Payloads/BoolPayload';
import { LogPayload } from './Payloads/LogPayload';
import { NullPayload } from './Payloads/NullPayload';
import { Payload } from './Payloads/Payload';

export class PayloadFactory {
    protected values: any[];

    protected static payloadFinder: Function | null = null;

    public static createForValues(args: any[]): any[] {
        return new this(args).getPayloads();
    }

    public static registerPayloadFinder(callable: Function) {
        this.payloadFinder = callable;
    }

    public constructor(values: any[]) {
        this.values = values;
    }

    public getPayloads(): any[] {
        return this.values.map(value => {
            return this.getPayload(value);
        });
    }

    protected getPayload(value: any): Payload {
        // if (this.payloadFinder) {
        //     if ($payload = (static::$payloadFinder)($value)) {
        //         return $payload;
        //     }
        // }

        if (typeof value === 'boolean') {
            return new BoolPayload(value);
        }

        if (value === null) {
            return new NullPayload();
        }

        // if ($value instanceof Carbon) {
        //     return new CarbonPayload($value);
        // }

        const primitiveValue = ArgumentConverter.convertToPrimitive(value);

        return new LogPayload(primitiveValue);
    }
}
