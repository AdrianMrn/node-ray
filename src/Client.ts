/* eslint-disable no-async-promise-executor */
//use Exception;
//use Spatie\Ray\Exceptions\StopExecutionRequested;

import { Request } from './Request';
import axios from 'axios';

export class Client {
    protected portNumber: number;
    protected host: string;

    public constructor(portNumber = 23517, host = 'localhost') {
        this.portNumber = portNumber;

        this.host = host;
    }

    public async send(request: Request) {
        await axios.post(`http://${this.host}:${this.portNumber}/`, request.toArray());

        //console.log(request.toJson());
        // TODO: send axios post request
    }

    public async lockExists(lockName: string) {
        return new Promise(async (resolve, reject) => {
            //console.log(lockName);

            const resp = await axios.get(`http://${this.host}:${this.portNumber}/locks/${lockName}`);

            //console.log('resp=', resp);

            //console.log(resp.data.stop_exectution);
            if (resp.data.stop_execution) {
                ///console.log(resp.data);

                reject(new Error('stopping execution'));

                //resolve(new Error('stopping execution'));
            }

            if (resp.data.stop_execution) {
                reject(new Error('stopping execution'));
            } else {
                if (typeof resp.data['active'] === 'undefined') {
                    console.log(resp.data);
                    resolve(resp.data);
                } else {
                    //resolve(resp.data);

                    //console.log(resp.data);
                    if (resp.data.stop_execution) {
                        //resolve(new Error('stopping execution'));
                        //return new Error('abc');
                        reject(new Error('test'));
                    } else {
                        resolve(resp.data);
                    }
                }
            }

            reject(new Error('test'));
        });
        //}//});

        /*
        curlHandle = this.getCurlHandleForUrl('get', `locks/${lockName}`);
        curlError = null;

        try {
            curlResult = curl_exec(curlHandle);

            if (curl_errno(curlHandle)) {
                curlError = curl_error(curlHandle);
            }

            if (curlError) {
                throw new Exception;
            }

            if (! curlResult) {
                return false;
            }

            response = json_decode(curlResult, true);

            if (response['stop_execution'] ?? false) {
                throw StopExecutionRequested::make();
            }

            return response['active'] ?? false;
        } catch (Exception exception) {
            if (exception instanceof StopExecutionRequested) {
                throw exception;
            }
        } finally {
            curl_close(curlHandle);
        }
        */

        //return false;
    }

    protected getCurlHandleForUrl(method: string, url: string) {
        return this.getCurlHandle(method, `http://${this.host}:${this.portNumber}/${url}`);
    }

    protected getCurlHandle(method: string, fullUrl: string) {
        console.log(method, fullUrl);
        /*
        curlHandle = curl_init();

        curl_setopt(curlHandle, CURLOPT_URL, fullUrl);

        curl_setopt(curlHandle, CURLOPT_HTTPHEADER, array_merge(['Accept: application/json', 'Content-Type: application/json']));

        curl_setopt(curlHandle, CURLOPT_USERAGENT, 'Ray 1.0');
        curl_setopt(curlHandle, CURLOPT_RETURNTRANSFER, true);
        curl_setopt(curlHandle, CURLOPT_TIMEOUT, 2);
        curl_setopt(curlHandle, CURLOPT_SSL_VERIFYPEER, true);
        curl_setopt(curlHandle, CURLOPT_HTTP_VERSION, CURL_HTTP_VERSION_1_1);
        curl_setopt(curlHandle, CURLOPT_ENCODING, '');
        curl_setopt(curlHandle, CURLINFO_HEADER_OUT, true);
        curl_setopt(curlHandle, CURLOPT_FAILONERROR, true);

        if (method === 'post') {
            curl_setopt(curlHandle, CURLOPT_POST, true);
        }

        return curlHandle;
        */
    }
}
