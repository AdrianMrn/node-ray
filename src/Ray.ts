/* eslint-disable no-undef */
/* eslint-disable no-useless-catch */

import md5 from 'md5';
import { v4 as uuidv4, v1 as uuidv1 } from 'uuid';
import { Mixin } from 'ts-mixer';
import { nonCryptoUuidV4, sleep } from './lib/utils';
import { ClearAllPayload } from './Payloads/ClearAllPayload';
import { Client } from './Client';
import { ColorPayload } from './Payloads/ColorPayload';
import { Counters } from './Support/Counters';
import { CreateLockPayload } from './Payloads/CreateLockPayload';
import { CustomPayload } from './Payloads/CustomPayload';
import { DecodedJsonPayload } from './Payloads/DecodedJsonPayload';
import { FileContentsPayload } from './Payloads/FileContentsPayload';
import { HideAppPayload } from './Payloads/HideAppPayload';
import { HidePayload } from './Payloads/HidePayload';
import { HtmlPayload } from './Payloads/HtmlPayload';
import { ImagePayload } from './Payloads/ImagePayload';
import { JsonStringPayload } from './Payloads/JsonStringPayload';
import { LogPayload } from './Payloads/LogPayload';
import { NewScreenPayload } from './Payloads/NewScreenPayload';
import { NotifyPayload } from './Payloads/NotifyPayload';
import { Payload } from './Payloads/Payload';
import { PayloadFactory } from './PayloadFactory';
import { RayColors } from './Concerns/RayColors';
import { RaySizes } from './Concerns/RaySizes';
import { RemovePayload } from './Payloads/RemovePayload';
import { Request } from './Request';
import { Settings } from './Settings/Settings';
import { SettingsFactory } from './Settings/SettingsFactory';
import { ShowAppPayload } from './Payloads/ShowAppPayload';
import { SizePayload } from './Payloads/SizePayload';
import { TablePayload } from './Payloads/TablePayload';
import { XmlPayload } from './Payloads/XmlPayload';
import randomInt from 'random-int';


export type BoolFunction = () => boolean;

export class Ray extends Mixin(RayColors, RaySizes) {

    protected static lockCounter = 0;

    public settings: Settings;

    protected static client: Client;

    public static counters: Counters = new Counters();

    public static fakeUuid: string;

    public uuid: string;

    // @var \Symfony\Component\Stopwatch\Stopwatch[]
    public static stopWatches: Record<string, unknown> = {};

    public static enabled = true;

    public static create(client: Client | null = null, uuid: string | null = null): Ray
    {
        const settings = SettingsFactory.createFromConfigFile();

        return new this(settings, client, uuid);
    }

    public constructor(settings: Settings, client: Client | null = null, uuid: string | null = null)
    {
        super();

        this.settings = settings;

        Ray.client = client ?? Ray.client ?? new Client(this.settings.port, this.settings.host);

        this.uuid = uuid ?? Ray.fakeUuid ?? nonCryptoUuidV4();

        Ray.enabled = this.settings.enable !== false;
    }

    public enable(): this
    {
        Ray.enabled = true;

        return this;
    }

    public disable(): this
    {
        Ray.enabled = false;

        return this;
    }

    public enabled(): boolean
    {
        return Ray.enabled;
    }

    public disabled(): boolean
    {
        return !Ray.enabled;
    }

    public static useClient(client: Client): void
    {
        this.client = client;
    }

    public newScreen(name = ''): this
    {
        const payload = new NewScreenPayload(name);

        return this.sendRequest(payload);
    }

    public clearAll(): this
    {
        const payload = new ClearAllPayload();

        return this.sendRequest(payload);
    }

    public clearScreen(): this
    {
        return this.newScreen();
    }

    public color(color: string): this
    {
        const payload = new ColorPayload(color);

        return this.sendRequest(payload);
    }

    public size(size: string): this
    {
        const payload = new SizePayload(size);

        return this.sendRequest(payload);
    }

    public remove(): this
    {
        const payload = new RemovePayload();

        return this.sendRequest(payload);
    }

    public hide(): this
    {
        const payload = new HidePayload();

        return this.sendRequest(payload);
    }

    public getOrigin(): any
    {
        // const frames = StackTrace.getSync().filter(frame =>
        // {
        //     return frame.getFileName().includes('Ray');
        // });


        // console.log(frames);

    }

    /**
     * @param string|callable stopwatchName
     *
     * @return this
     *
    public measure(stopwatchName = 'default'): this
    {
        if (stopwatchName instanceof Closure) {
            return this.measureClosure(stopwatchName);
        }

        if (typeof this.stopWatches[stopwatchName] === 'undefined') {
            stopwatch = new Stopwatch(true);
            this.stopWatches[stopwatchName] = stopwatch;

            event = stopwatch.start(stopwatchName);
            const payload = new MeasurePayload(stopwatchName, event);
            payload.concernsNewTimer();

            return this.sendRequest(payload);
        }

        stopwatch = this.stopWatches[stopwatchName];
        event = stopwatch.lap(stopwatchName);
        const payload = new MeasurePayload(stopwatchName, event);

        return this.sendRequest(payload);
    }

    public trace(?Closure startingFromFrame = null): this
    {
        backtrace = Backtrace::create();

        if (class_exists(LaravelRay::class) && function_exists('base_path')) {
            backtrace.applicationPath(base_path());
        }

        if (startingFromFrame) {
            backtrace.startingFromFrame(startingFromFrame);
        }

        const payload = new TracePayload(backtrace.frames());

        return this.sendRequest(payload);
    }

    public backtrace(?Closure startingFromFrame = null): this
    {
        return this.trace(startingFromFrame);
    }

    public caller(): this
    {
        backtrace = Backtrace::create();

        payload = (new CallerPayload(backtrace.frames()));

        return this.sendRequest(payload);
    }

    protected measureClosure(Closure closure): this
    {
        stopwatch = new Stopwatch(true);

        stopwatch.start('closure');

        closure();

        event = stopwatch.stop('closure');

        const payload = new MeasurePayload('closure', event);

        return this.sendRequest(payload);
    }

    */

    public stopTime(stopwatchName = ''): this
    {
        if (stopwatchName === '') {
            Ray.stopWatches = {};

            return this;
        }

        if (typeof Ray.stopWatches[stopwatchName] !== 'undefined') {
            delete Ray.stopWatches[stopwatchName];
        }

        return this;
    }

    public notify(text: string): this
    {
        const payload = new NotifyPayload(text);

        return this.sendRequest(payload);
    }

    public toJson(...values: any[]): this
    {
        const payloads = values.map(value =>
        {
            return new JsonStringPayload(value);
        });

        return this.sendRequest(payloads);
    }

    public json(...jsons: string[]): this
    {
        const payloads = jsons.map(json =>
        {
            return new DecodedJsonPayload(json);
        });

        return this.sendRequest(payloads);
    }

    public file(filename: string): this
    {
        const payload = new FileContentsPayload(filename);

        return this.sendRequest(payload);
    }

    public image(location: string): this
    {
        return this.sendRequest(new ImagePayload(location));
    }

    public die(status = ''): void
    {
        if (status.length) {
            console.error(status);
        }

        process.exit(-1);
    }

    public className(object: any): this
    {
        return this.send(object.constructor.name);
    }

    public showWhen(booleanOrCallable: boolean | BoolFunction): this
    {
        if (typeof booleanOrCallable === 'function') {
            booleanOrCallable = (booleanOrCallable as BoolFunction)();
        }

        if (!booleanOrCallable) {
            this.remove();
        }

        return this;
    }

    public showIf(booleanOrCallable: boolean | BoolFunction): this
    {
        return this.showWhen(booleanOrCallable);
    }

    public removeWhen(booleanOrCallable: boolean | BoolFunction): this
    {
        if (typeof booleanOrCallable === 'function') {
            booleanOrCallable = (booleanOrCallable as BoolFunction)();
        }

        if (booleanOrCallable) {
            this.remove();
        }

        return this;
    }

    public removeIf(booleanOrCallable: boolean | BoolFunction): this
    {
        return this.removeWhen(booleanOrCallable);
    }

    public ban(): this
    {
        return this.send('ðŸ•¶');
    }

    public charles(): this
    {
        return this.send('ðŸŽ¶ ðŸŽ¹ ðŸŽ· ðŸ•º');
    }

    public table(values: any[], label = 'Table'): this
    {
        const payload = new TablePayload(values, label);

        return this.sendRequest(payload);
    }

    public count(name: string | null = null): this
    {
        const fingerprint = ''; //(new DefaultOriginFactory()).getOrigin().fingerPrnumber();

        const [ray, times] = Ray.counters.increment(name || fingerprint);

        let message = `Called `;

        if (name) {
            message += `'${name}' `;
        }

        message += `${times} ${times === 1 ? 'time' : 'times'}.`;

        ray.sendCustom(message, 'Count');

        return ray;
    }

    public clearCounters(): this
    {
        Ray.counters.clear();

        return this;
    }

    public async pause()
    {
        Ray.lockCounter++;

        const lockName = md5(`${new Date().getTime()}-${Ray.lockCounter}`);
        const payload = new CreateLockPayload(lockName);

        this.sendRequest(payload);

        let exists: any;

        do {
            sleep(1);
            try {
                exists = await Ray.client.lockExists(lockName);
            } catch (err) {
                return false;
            }

            if (exists !== true && exists && exists.stop_execution) {
                return false;
            }
        } while (exists.active);

        return this;
    }

    public xml(xml: string): this
    {
        const payload = new XmlPayload(`<a>${xml}</a>`);

        return this.sendRequest(payload);
    }

    public html(html = ''): this
    {
        const payload = new HtmlPayload(html);

        return this.sendRequest(payload);
    }

    public raw(...args: any[]): this
    {
        if (!args.length) {
            return this;
        }

        const payloads = args.map(argument =>
        {
            return LogPayload.createForArguments([argument]);
        });

        return this.sendRequest(payloads);
    }

    public send(...args: any[]): this
    {
        if (!args.length) {
            return this;
        }

        if (this.settings.always_send_raw_values) {
            return this.raw(...args);
        }


        const payloads = PayloadFactory.createForValues(args);

        return this.sendRequest(payloads);
    }

    public pass(argument: any): any
    {
        this.send(argument);

        return argument;
    }

    public showApp(): this
    {
        const payload = new ShowAppPayload();

        return this.sendRequest(payload);
    }

    public hideApp(): this
    {
        const payload = new HideAppPayload();

        return this.sendRequest(payload);
    }

    public sendCustom(content: string, label = ''): this
    {
        const payload = new CustomPayload(content, label);

        return this.sendRequest(payload);
    }

    public sendRequest(payloads: Payload | Payload[], meta: any[] = []): this
    {
        if (!this.enabled()) {
            return this;
        }

        if (!Array.isArray(payloads)) {
            payloads = [payloads];
        }

        const allMeta = Object.assign({}, {
            node_version: process.versions.node
        }, meta);

        payloads.forEach(payload =>
        {
            payload.remotePath = this.settings.remote_path;
            payload.localPath = this.settings.local_path;
        });

        const request = new Request(this.uuid, payloads, allMeta);

        Ray.client?.send(request);

        return this;
    }
}

export const ray = (...args: any[]) =>
{
    return Ray.create().send(...args);
};
