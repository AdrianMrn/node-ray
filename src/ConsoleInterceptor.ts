import { Ray } from './Ray';

export class ConsoleInterceptor {
    public static active = false;
    public static consoleLog: any;

    public intercept() {
        ConsoleInterceptor.active = true;
        ConsoleInterceptor.consoleLog = console.log;

        console.log = this.wrapper;
    }

    public reset() {
        ConsoleInterceptor.active = false;

        console.log = ConsoleInterceptor.consoleLog;
    }

    public active(): boolean {
        return ConsoleInterceptor.active;
    }

    protected wrapper(...args: any[]) {
        if (typeof Ray.client !== 'undefined' && Ray.client.isRayAvailable()) {
            Ray.create().send(...args);
        }

        ConsoleInterceptor.consoleLog(...args);
    }
}
