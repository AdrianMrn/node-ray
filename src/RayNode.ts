import { existsSync } from 'fs';
import { Client } from './Client';
import { FileContentsPayload } from './Payloads/FileContentsPayload';
import { ImagePayload } from './Payloads/ImagePayload';
import { Ray as BaseRay } from './Ray';
import { SettingsFactory } from './Settings/SettingsFactory';

export class Ray extends BaseRay
{

    public static create(client: Client | null = null, uuid: string | null = null): Ray
    {
        const settings = SettingsFactory.createFromConfigFile();

        return new this(settings, client, uuid);
    }

    public die(status = ''): void
    {
        if (status.length) {
            console.error(status);
        }

        process.exit(-1);
    }

    public file(filename: string): this
    {
        const payload = new FileContentsPayload(filename);

        return this.sendRequest(payload);
    }

    public image(location: string): this
    {
        if (existsSync(location)) {
            location = `file://${location}`;
        }

        const payload = new ImagePayload(location);

        return this.sendRequest(payload);
    }
}

export const ray = (...args: any[]) =>
{
    return Ray.create().send(...args);
};
