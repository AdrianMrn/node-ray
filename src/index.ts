import { Ray, ray } from './Ray';

export { Ray, ray } from './Ray';

export default ray;
