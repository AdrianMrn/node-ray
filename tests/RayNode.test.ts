/* eslint-disable no-undef */
/* eslint-disable no-useless-catch */

import { Ray as RayNode } from './../src/RayNode';
import { FakeClientNode } from './TestClasses/FakeClientNode';

let client: FakeClientNode, myRay: RayNode;

beforeEach(() => {
    RayNode.defaultSettings = {
        enable: true,
        host: 'localhost',
        port: 23510,
        local_path: null,
        remote_path: null,
        always_send_raw_values: false,
        not_defined: false,
    };

    client = new FakeClientNode();
    myRay = RayNode.create(client, 'fakeUuid');
    myRay.clearCounters();
});

it('sends an existing file payload', () => {
    myRay.file(`${__dirname}/TestData/test.txt`);

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends a missing file payload', () => {
    myRay.file('missing.txt');

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends an existing image file payload', () => {
    myRay.image(`${__dirname}/TestData/test.txt`);

    expect(client.sentPayloads()).toMatchSnapshot();
});

it('sends an image url payload', () => {
    myRay.image(`http://localhost:35599/test.png`);

    expect(client.sentPayloads()).toMatchSnapshot();
});
