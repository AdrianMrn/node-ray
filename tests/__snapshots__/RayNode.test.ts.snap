// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sends a missing file payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "File not found: '{missing.txt'",
          "label": "File",
        },
        "origin": Object {
          "file": "/test/file.js",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an existing file payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "one,two,three<br>",
          "label": "test.txt",
        },
        "origin": Object {
          "file": "/test/file.js",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an existing image file payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "<img src=\\"file:///tests/TestData/test.txt\\" alt=\\"\\" />",
          "label": "Image",
        },
        "origin": Object {
          "file": "/test/file.js",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;

exports[`sends an image url payload 1`] = `
Array [
  Object {
    "meta": Array [],
    "payloads": Array [
      Object {
        "content": Object {
          "content": "<img src=\\"http://localhost:35599/test.png\\" alt=\\"\\" />",
          "label": "Image",
        },
        "origin": Object {
          "file": "/test/file.js",
          "function_name": "xxxx",
          "line_number": 999,
        },
        "type": "custom",
      },
    ],
    "uuid": "fakeUuid",
  },
]
`;
